code_challenges/adhoc/ad_hock_challenge.py

```diff
@@ -2,18 +2,17 @@
 # The dictionaries keys will be strings and the values will be functions.
 # The return value should be the return value from the function from the input dict,  at the input string, 
 # when called with the values in the input iterable as arguments.
-def find_the_function(functions, function_name, the_numbers):
-    calling = functions[function_name]
-    return calling(the_numbers)
+def find_the_function(functions, function_name, values):
+    # calling = functions[function_name]
+    return functions[function_name](*values)#calling(the_numbers)
     
-def adding(the_numbers):
-    total = 0
-    for number in the_numbers:
-        total += number
-    return total
+def adding(*nums):
+    
+      
+    return sum(nums)
 
 
-def even_it_up(the_numbers):
+def even_it_up(*the_numbers): # *mutiple arguments
```

Comment:
If you want to keep comments like this for later study, I'd suggest creating a notes repo with examples of usage.

Remove unneeded commnets.